api/hypershift/v1beta1/openstack.go

```diff
@@ -136,6 +136,21 @@ type OpenStackPlatformSpec struct {
 	// +kubebuilder:validation:XValidation:rule="isIP(self)",message="floatingIP must be a valid IPv4 or IPv6 address"
 	// +optional
 	IngressFloatingIP string `json:"ingressFloatingIP,omitempty"`
+
+	// SkipCleanupORCImageResources is a field that determines whether the OpenStack image
+	// associated with an ORC (OpenStack Resource Controller) image should be removed when
+	// deleting the ORC image.
+	// By default, ORC deletes the OpenStack image when the ORC image is deleted,
+	// which happens when the HostedCluster is deleted.
+	//
+	// If set to `true`, the OpenStack Image will not be removed, allowing it to be
+	// reused by other HostedClusters. This is useful in cases where images are shared
+	// across multiple clusters or in environments where we want to avoid unnecessary downloads
+	// for each test cluster, saving time and cost.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="SkipCleanupORCImageResources is immutable"
+	// +optional
+	SkipCleanupORCImageResources *bool `json:"SkipCleanupORCImageResources,omitempty"`
```

Comment:
@JoelSpeed would tell us off for using a boolean here, btw. The wonks are currently telling us that booleans should generally be enums instead.

This is what we do in ORC, in fact. `onDelete` is one of `delete` or `detach`: https://k-orc.cloud/crd-reference/#ondelete. This leaves open the possibility of other behaviours in the future without requiring a backwards-incompatible change.

Reply:
I thought about it too but I didn't find the user experience so great, in that particular case; but maybe it's fine as long as it's well documented?

Reply:
^ Just FYI on OpenShift API conventions in case you didn't have it. Joel shared that with me in the recent past.

Reply:
An `ImageRetentionPolicy: Prune | Orphan` might work here

Reply:
I like that better than what we have in ORC, btw (which iirc I stole from ASO). Meh, it's baked now. Maybe we'll update it if we bump the api version.

Reply:
Good idea Joel. I was hesitant to that because it's an abstraction to what ORC does, but I talked to Matt and he seems fine with it, so am I.